### PR TITLE
Update contributing instructions

### PR DESCRIPTION
**Pull request by Mackiovello**
For starcounter/guidelines#17

Should instructions regarding testing be included as well? 

I'm thinking that it might be useful to combine these instructions with the [branching instructions](https://starcounter.gitbooks.io/guidelines/content/branching.html)

### CODE REVIEW

**Comment by warpech** [comment]
> Should instructions regarding testing be included as well?

Yes, for the apps that have tests.

> I'm thinking that it might be useful to combine these instructions with the [branching instructions](https://starcounter.gitbooks.io/guidelines/content/branching.html)

Indeed, but as agreed - there is just to much information to maintain in a DRY way. So I propose that most of the instructions are moved to Guidelines.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,48 +1,49 @@
...
 ## How to build and run
 
 ### Build and run using CLI
 
-1. Check out StarcounterSamples/Products repo from GitHub
-2. Build the solution using `tools\build.bat`, in Windows, or executing `cmd //c "tools\build.bat"` in Git Bash
-3. Run it by launching `tools\run.bat` in Windows, or executing `cmd //c "tools\run.bat"` in Git Bash
+1. Clone this repo to your local machine
+2. Build the solution using `build.bat`, in Windows, or executing `./build.bat` in Git Bash
+3. Run it by launching `run.bat` in Windows, or executing `./run.bat"` in Git Bash
```
Comment (warpech) [inline CONTRIBUTING.md]: Amazing! I was totally not aware that I can just run bat scripts with a `./` in Git Bash. @tomalec did you know?

**Comment by warpech** [comment]
>> Should instructions regarding testing be included as well?

> Yes, for the apps that have tests.

On the other hand, these Node tests are really poor and just should be removed in my opinion. So there is no way to rewrite the instructions for those.

**Comment by Mackiovello** [comment]
> On the other hand, these Node tests are really poor and just should be removed in my opinion. So there is no way to rewrite the instructions for those.

Created an issue for that here: https://github.com/StarcounterApps/Products/issues/42

File: CONTRIBUTING.md (hunk 1 above)
Comment (tomalec) [inline CONTRIBUTING.md]: nope 😌

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,27 +31,28 @@
...
+6. Package the solution using `package.bat` in Windows, or executing `./package.bat` in Git Bash
+7. Publish the zip file found in the `src/[app]` directory to the App Warehouse:
+    * Go to [the warehouse](http://warehouse.starcounter.com/)
+    * Login with your credentials
+    * Click "Upload version..." on the page of the app to release
+    * Verify that the package was uploaded
+8. Commit the incremented version with a message identical to the version:
     ```git
-    git add .
+    git add -u // -u flag to ensure that the zip is not included
```
Comment (warpech) [inline CONTRIBUTING.md]: Nice git-fu!

**Comment by warpech** [comment]
@Mackiovello you have removed a section, then added it back. Any rationale for that?

I believe that we should not have a direct link to RebelsLounge, because that is a private repo.

We should have a direct link to Guidelines GitBook, because that one is publicly available.

The `CONTRIBUTING.md` file does not have to copy the text from Guidelines, it can just be the entry point to discover relevant sections of it.

**Comment by Mackiovello** [comment]
> @Mackiovello you have removed a section, then added it back. Any rationale for that?

I have looked through the commits and tried to figure out which one you mean. I'm not finding it.

> I believe that we should not have a direct link to RebelsLounge, because that is a private repo.

That's right. I would suggest moving the release instructions to the Guidelines. No reason for them to be in a private repo.

I think the right structure to have is to move the build/run/test instructions to the `README` file, which essentially consists of using the batch files. There would then be nothing but two links in the `CONTRIBUTING` file that points to instructions on how to contribute code and how to release a package in the Guidelines GitBook.

WDYT?

**Comment by warpech** [comment]
> I have looked through the commits and tried to figure out which one you mean. I'm not finding it.

Sorry, I was wrong. You have removed "How to release a package" and added "Contributing Code".

> There would then be nothing but two links in the `CONTRIBUTING` file that points to instructions on how to contribute code and how to release a package in the Guidelines GitBook.

Sounds good! Also, optionally in some repos there will be instructions how to test.

**Comment by Mackiovello** [comment]
Releasing to warehouse instructions are now moved to the [guidelines](https://starcounter.gitbooks.io/guidelines/content/releasing-to-warehouse.html) together with the [contributing code](https://starcounter.gitbooks.io/guidelines/content/contributing-code.html) instructions.

I would consider this one done now. The next step is to apply this to the other apps. 

Is there anything more that should be changed before merging?

File: CONTRIBUTING.md (hunk 1 above)
Comment (warpech) [inline CONTRIBUTING.md]: Just one more note. In the points 2 and 3 there are some unneccessary commas. Also we could decide on wording: using, launching, executing.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,48 +1,26 @@
...
 
-1. Check out StarcounterSamples/Products repo from GitHub
-2. Open `Products.proj` in Visual Studio
+1. Clone this repo to your local machine
+2. Open the `.sln` file in Visual Studio
 3. Build and run it using Debug in Visual Studio (**Debug** > **Start debugging** or <kbd>F5</kbd>)
 4. Go to [http://localhost:8080/products](http://localhost:8080/products)
 
-## How to test
+## Contributing Code
```
Comment (warpech) [inline CONTRIBUTING.md]: "code" should be lowercase to be consistent with the other headers.

(or make other headers capitalised, too?)

Comment (Mackiovello) [inline CONTRIBUTING.md]: Both your comments are addressed in 	a6e599c. 

> Also we could decide on wording: using, launching, executing.

I favor executing

**Pull request merged**